### DISCUSSION TITLE
I am curious how these KEYS are found during the reverse engineering process

### DISCUSSION THREAD

**Discussion by FoxMcWeezer** [cs0x7f/cstimer#345] [Q&A] [answered]
https://github.com/cs0x7f/cstimer/blob/master/src/js/bluetooth.js#L230

I would imagine these keys are kept top-secret by the manufacturer. How did you find these? @cs0x7f. Thanks

**Comment by cs0x7f** [comment, marked as answer]
These were not what I found, it was copied from some other places (such as cubeast, cubing.js, etc.)
In addition, some manufacturers may disclose specifications to some software developers to expand their sales of their hardwares, since whether it can connect with third-party tools is one of the considerations of many cuber to buy smart cubes/timers.
